spatialprofilingtoolbox/db/scripts/list_studies.py

```diff
@@ -5,9 +5,10 @@
 from spatialprofilingtoolbox.workflow.common.cli_arguments import add_argument
 from spatialprofilingtoolbox.standalone_utilities.module_load_error import SuggestExtrasException
 try:
-    from spatialprofilingtoolbox import DatabaseConnectionMaker  # pylint: disable=ungrouped-imports
+    from spatialprofilingtoolbox import DatabaseConnectionMaker
 except ModuleNotFoundError as e:
     SuggestExtrasException(e, 'db')
+from spatialprofilingtoolbox import DatabaseConnectionMaker
```

Comment:
So the way it was supposed to work is that the CLI tools for different major submodules (`db`, `workflow`, etc.) were supposed to tell a helpful suggestion in case they are used without the corresponding `pip install spatialprofilingtoolbox[<module name>]`, i.e. if some dependencies are missing. Rather than just triggering the normal unhelpful "import fails" error.
This functionality was sort of tricky to implement without creating linting errors. If the import is in a try catch block, later blocks would complain that the symbol might not be bound.
Today I discovered that I can just use the try block system, then immediately import like normal (not in the try block). This satisfies the later code blocks that use the imported symbols, but doesn't break the "suggestion" functionality.

Reply:
Interesting. Why wouldn't it just throw the uncaught import error in practice?

Reply:
I looked a little closer, and apparently the way I implemented it was for the original exception to be re-raised. So the ugly error still happens, just after a helpful message.
(The ugly error happens inside SuggestExtrasException's init, so the second occurrence of the import line isn't actually running in such cases.)

Reply:
Here's how it looks (example: `spt db status` with pandas missing):

![image](https://github.com/nadeemlab/SPT/assets/18318706/2c88d451-2a05-474b-8480-5f0c3dd4587c)